src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs

```diff
@@ -1344,40 +1344,57 @@ private void WriteMessageAligned(string message, bool prefixAlreadyWritten, int
             {
                 int adjustedPrefixWidth = _prefixWidth + prefixAdjustment;
 
-                // The string may contain new lines, treat each new line as a different string to format and send to the console
-                string[] nonNullMessages = SplitStringOnNewLines(message);
-                for (int i = 0; i < nonNullMessages.Length; i++)
+                if (Traits.Instance.EscapeHatches.DoNotOptimizeConsoleLogger || !ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_0))
                 {
-                    string nonNullMessage = nonNullMessages[i];
-                    // Take into account the new line char which will be added to the end or each reformatted string
-                    int bufferWidthMinusNewLine = _bufferWidth - 1;
-
-                    // If the buffer is larger then the prefix information (timestamp and key) then reformat the messages. 
-                    // If there is not enough room just print the message out and let the console do the formatting
-                    bool bufferIsLargerThanPrefix = bufferWidthMinusNewLine > adjustedPrefixWidth;
-                    bool messageAndPrefixTooLargeForBuffer = (nonNullMessage.Length + adjustedPrefixWidth) > bufferWidthMinusNewLine;
-                    if (bufferIsLargerThanPrefix && messageAndPrefixTooLargeForBuffer && _alignMessages)
+                    // The string may contain new lines, treat each new line as a different string to format and send to the console
+                    string[] nonNullMessages = SplitStringOnNewLines(message);
```

Comment:
Can you use Span rather than creating more strings here?

Reply:
OLD code used in escape hatch to avoid regressions. Kept as close as possible to previous version.